Test/inline/testWhereExpr.bpl.expect

```diff
@@ -0,0 +1,59 @@
+
+procedure a();
+
+
+
+implementation a()
+{
+
+  anon0:
+    call b();
+    return;
+}
+
+
+
+procedure {:inline 1} b();
+
+
+
+implementation {:inline 1} b()
+{
+  var v: int where v > 3;
+
+  anon0:
+    havoc v;
+    return;
+}
+
+
+after inlining procedure calls
+procedure a();
+
+
+implementation a()
+{
+  var inline$b$0$v: int where inline$b$0$v > 3;
+
+  anon0:
+    goto inline$b$0$Entry;
+
+  inline$b$0$Entry:
+    havoc inline$b$0$v /* where inline$b$0$v > 3 */;
+    assume inline$b$0$v > 3;
+    goto inline$b$0$anon0;
+
+  inline$b$0$anon0:
+    havoc inline$b$0$v /* where inline$b$0$v > 3 */;
+    goto inline$b$0$Return;
```

Comment:
Strange! Why is an assume not injected here?

Reply:
I am not sure. My suspicion is that an assumption would be redundant both here and above but I have not fully convinced myself yet. Empirically, it seems that `havoc` must respect the type constraint by default, e.g. the code below verifies. I will make another PR if I figure it out.

```Boogie
procedure b() {
    var d:int where d > 3;
    havoc d;
    assert d > 3;
}
```

Reply:
Here is a conjecture. 

- It is indeed redundant to put the assumes after the havocs in the inlining code because some processing later on (possibly VC generation) takes care of handling the where clauses after havoc appropriately.
- The bug in the inlining code you uncovered turned off the treatment above because the where clause was not appropriately substituted.
- The bug was discovered by somebody who did not understand the root cause and chose to fix the problem by putting the assumes in the inlining code itself.

If my conjecture is correct, then after your fix has landed, it ought to be possible to delete the assume injection. Of course, we have to make sure that the regression tests are adequate to check the behavior. You might need to add more tests.

I can approve your PR if you are ready to land it.

Reply:
Thank you, @shazqadeer! I think you are right, let me do some testing to make sure this is the case and then make another PR about the `assume`. I am ready to land this one!